### PR TITLE
Add Basis of preparation controller and supporting tests and template

### PR DESCRIPTION
**Pull request by BSaunders95**
Add Basis of preparation controller and supporting tests and template

Resolves SFA-628

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationController.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                return getTemplateName();
+            }
+        } catch (ServiceException e) {
+
+            LOGGER.errorRequest(request, e.getMessage(), e);
+            return ERROR_VIEW;
+        }
+
+        return Navigator.getNextControllerRedirect(this.getClass(), companyNumber, transactionId, companyAccountsId);
+
```
Comment (ch-mcabrera) [inline src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationController.java]: empty line ;)

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    private static final String BASIS_OF_PREPARATION_PATH = "/company/" + COMPANY_NUMBER +
+                                                            "/transaction/" + TRANSACTION_ID +
+                                                            "/company-accounts/" + COMPANY_ACCOUNTS_ID +
+                                                            "/small-full/basis-of-preparation";
+
+    private static final String REVIEW_PATH = "/company/" + COMPANY_NUMBER +
+                                                "/transaction/" + TRANSACTION_ID +
+                                                "/company-accounts/" + COMPANY_ACCOUNTS_ID +
+                                                "/small-full/review";
+
```
Comment (ch-mcabrera) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationControllerTests.java]: we could create a constant for SMALL_FULL_PATH 

```
private static final String SMALL_FULL_PATH = "/company/" + COMPANY_NUMBER +
        "/transaction/" + TRANSACTION_ID +
        "/company-accounts/" + COMPANY_ACCOUNTS_ID +
        "/small-full";

private static final String BASIS_OF_PREPARATION_PATH = SMALL_FULL_PATH + "/basis-of-preparation"
private static final String REVIEW_PATH  =  SMALL_FULL_PATH + "/review"
```

Comment (BSaunders95) [reply src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationControllerTests.java]: Added 👍

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationController.java (hunk 1 above)
Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BasisOfPreparationController.java]: Removed ;)

**Pull request merged**